File: src/main/java/dev/foxgirl/torrent/bencode/BencodeDecodeException.java
package dev.foxgirl.torrent.bencode;

import java.io.IOException;
import java.io.Serial;

public class BencodeDecodeException extends IOException {

    @Serial
    private static final long serialVersionUID = -7444516590442325786L;

    public BencodeDecodeException() {
    }

    public BencodeDecodeException(String message) {
        super(message);
    }

    public BencodeDecodeException(String message, Throwable cause) {
        super(message, cause);
    }

    public BencodeDecodeException(Throwable cause) {
        super(cause);
    }

}

File: src/main/kotlin/dev/foxgirl/torrent/bencode2/BencodeDecodeException.kt
package dev.foxgirl.torrent.bencode2

import java.io.IOException

/** Thrown when [BencodeDecoder] encounters invalid Bencode. */
class BencodeDecodeException : IOException {

    constructor() : super()
    constructor(message: String) : super(message)
    constructor(message: String, cause: Throwable) : super(message, cause)
    constructor(cause: Throwable) : super(cause)

}